src/data/stream/init.lean

```diff
@@ -43,6 +43,8 @@ funext (λ i, begin unfold drop, rw nat.add_assoc end)
 
 theorem nth_succ (n : nat) (s : stream α) : nth s (succ n) = nth (tail s) n := rfl
 
+@[simp] lemma nth_succ_cons (n : nat) (s : stream α) (x : α) : nth (x :: s) n.succ = nth s n := rfl
```

Comment:
Let's not aggravate the problem:
```suggestion
@[simp] lemma nth_cons_succ (x : α) (s : stream α) (n : ℕ) : nth (x :: s) n.succ = nth s n := rfl
```
Argument order matches `list.nthd_cons_succ`. Also, what's up with this file using `nat` instead of `ℕ`?

Reply:
This file has `nat` because it came from core lean. I want to leave it consistent so that a later chore rename will be easier.